test/fixtures/rules/std_rule_cases/L070.yml

```diff
@@ -0,0 +1,482 @@
+rule: L070
```

Comment:
I think I found a bug in nested CTE handling. This query should report an issue but does not:
```
WITH y1 AS (
    WITH x AS (
        SELECT a, b FROM MyTable
    )
    SELECT * FROM x
), y2 AS (
    WITH x AS (
        SELECT a, b, c FROM MyTable
    )
    SELECT * FROM x
)
SELECT * FROM y1
UNION ALL
SELECT * FROM y2
```

This similar query where I rename the second `x` CTE as `x2` **does** report an issue:
```
WITH y1 AS (
    WITH x AS (
        SELECT a, b FROM MyTable
    )
    SELECT * FROM x
), y2 AS (
    WITH x2 AS (
        SELECT a, b, c FROM MyTable
    )
    SELECT * FROM x2
)
SELECT * FROM y1
UNION ALL
SELECT * FROM y2
```

Reply:
BTW, I think rule L044 handles this case correctly. If you see opportunities to improve `SelectCrawler` to make it easier to handle this case, that could be worthwhile. This class has been useful but is still a work in progress.

Reply:
Yes, I missed this case; I'll dig into whether there's a better way to handle finding the ctes in the SelectCrawler

Reply:
@barrywhart I added a method in the SelectCrawler to build out the parent query in cases where the parent stack is added as an argument. This helps find ctes in the parent query - via the query lookup_ctes method - that are out of the scope of the segment with the set expression. Do you think this is a reasonable solution in this case?
https://github.com/sqlfluff/sqlfluff/pull/4028/files#diff-352afddb37ded0c1b8d8e0d6b5c4116308ec37b157d98bd3453eab3174e3b314R440